crates/searchers/openapi.yaml

```diff
@@ -0,0 +1,584 @@
+        - sellAmount
+        - buyAmount
+        - feeAmount
+        - side
+        - partiallyFillable
+        - class
+        - reward
+      properties:
+        uid:
+          $ref: "#/components/schemas/OrderUid"
+        sellToken:
+          $ref: "#/components/schemas/Address"
+        buyToken:
+          $ref: "#/components/schemas/Address"
+        sellAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        buyAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        feeAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        side:
+          $ref: "#/components/schemas/OrderSide"
+        partiallyFillable:
+          type: boolean
+        class:
+          $ref: "#/components/schemas/OrderClass"
+        reward:
+          type: number
+          description: |
+            The rewards, in COW base units, that would be given by the protocol
+            for settling this order.
+
+    ConstantProductPool:
+      description: |
+        A UniswapV2-like constant-product 2-token liquidity pool.
+      type: object
+      required:
+        - kind
+        - reserves
+        - fee
+      properties:
+        kind:
+          type: string
+          enum: [constantproduct]
+        reserves:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/TokenAmount"
+          description: |
+            A mapping of token address to reserve amounts.
+        fee:
+          $ref: "#/components/schemas/Decimal"
+
+    WeightedProductPool:
+      description: |
+        A Balancer-like weighted-product N-token liquidity pool.
+      type: object
+      required:
+        - kind
+        - reserves
+        - fee
+      properties:
+        kind:
+          type: string
+          enum: [weightedproduct]
+        reserves:
+          type: object
+          additionalProperties:
+            type: object
+            properties:
+              balance:
+                $ref: "#/components/schemas/TokenAmount"
+              weight:
+                $ref: "#/components/schemas/Decimal"
+          description: |
+            A mapping of token address to token balance and weight.
+        fee:
+          $ref: "#/components/schemas/Decimal"
+
+    StablePool:
+      description: |
+        A Curve-like stable N-token stable pool.
+      type: object
+      required:
+        - kind
+        - reserves
+        - amplificationParameter
+        - fee
+      properties:
+        kind:
+          type: string
+          enum: [stable]
+        reserves:
+          type: object
+          additionalProperties:
+            type: object
+            properties:
+              balance:
+                $ref: "#/components/schemas/TokenAmount"
+              scalingRate:
+                $ref: "#/components/schemas/BigInt"
+          description: |
+            A mapping of token address to token balance and scaling rate.
+        amplificationParameter:
```

Comment:
This is what it is called in the pool (also in the Curve whitepaper AFAIK). I just chose to use `amplicationParameter` because that is the established term for parameter in these kinds of pools.